.github/workflows/ci_workflow.yml

```diff
@@ -0,0 +1,397 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+
+name: ci
+
+on:
+  # Note: Jobs are skipped when ONLY Markdown (*.md) files are changed
+  push:
+    branches:
+      - master
+      - RB-*
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+      - RB-*
+    paths-ignore:
+      - '**.md'
+
+  schedule:
+    # Full nightly build
+    - cron: "0 0 * * *"
+
+jobs:
+  # Linux jobs run in Docker containers, so the latest OS version is OK. macOS 
+  # and Windows jobs need to be locked to specific virtual environment 
+  # versions to mitigate issues from OS updates, and will require maintenance 
+  # as OS versions are retired.
+
+  # ---------------------------------------------------------------------------
+  # Linux
+  # ---------------------------------------------------------------------------
+  # Note: 'runs-on' defines the GH-hosted VM. 'container' is the Docker 
+  #       container where the build occurs.
+  # Note: 'container' images are hosted on DockerHub
+
+  linux:
+    name: Linux CentOS 7 VFX CY2019 ${{ matrix.build-desc }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        build: [1, 2, 3, 4, 5, 6]
+        include:
+          - build: 1
+            build-desc: ''
+            build-type: Release
+            python-version: 3.7
+            shared: 'ON'
+            use-sse: 'ON'
+            cxx-compiler: g++
+            cc-compiler: gcc
+          - build: 2
+            build-desc: (Debug)
+            build-type: Debug
+            python-version: 3.7
+            shared: 'ON'
+            use-sse: 'ON'
+            cxx-compiler: g++
+            cc-compiler: gcc
+          - build: 3
+            build-desc: (static, no SSE, Python 2.7)
+            build-type: Release
+            python-version: 2.7
+            shared: 'OFF'
+            use-sse: 'OFF'
+            cxx-compiler: g++
+            cc-compiler: gcc
+          - build: 4
+            build-desc: '(Clang 7)'
```

Comment:
Are the  '  needed for the description?

Reply:
I overhauled the description handling so it's much more explicit now